com.unity.robotics.urdf-importer/Runtime/Extensions/UrdfGeometryVisual.cs

```diff
@@ -0,0 +1,111 @@
+﻿/*
+© Siemens AG, 2018
+Author: Suzannah Smith (suzannah.smith@siemens.com)
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+<http://www.apache.org/licenses/LICENSE-2.0>.
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using System;
+using UnityEngine;
+using UnityMeshImporter;
+
+namespace RosSharp.Urdf
+{
+    public class UrdfGeometryVisual : UrdfGeometry
+    {
+        public static void Create(Transform parent, GeometryTypes geometryType, Link.Geometry geometry = null)
+        {
+            GameObject geometryGameObject = null;
+
+            switch (geometryType)
+            {
+                case GeometryTypes.Box:
+                    geometryGameObject = GameObject.CreatePrimitive(PrimitiveType.Cube);
+                    geometryGameObject.transform.DestroyImmediateIfExists<BoxCollider>();
+                    break;
+                case GeometryTypes.Cylinder:
+                    geometryGameObject = GameObject.CreatePrimitive(PrimitiveType.Cylinder);
+                    geometryGameObject.transform.DestroyImmediateIfExists<CapsuleCollider>();
+                    break;
+                case GeometryTypes.Sphere:
+                    geometryGameObject = GameObject.CreatePrimitive(PrimitiveType.Sphere);
+                    geometryGameObject.transform.DestroyImmediateIfExists<SphereCollider>();
+                    break;
+                case GeometryTypes.Mesh:
+                        if (geometry != null)
+                        {
+                            geometryGameObject = CreateMeshVisual(geometry.mesh);
+                        }
+                        //else, let user add their own mesh gameObject
```

Comment:
I guess we won't let users change this script. How can users add their own mesh gameObject if the `geometry` variable is `null`?

Reply:
Yes, not in the script. This means that in the unity editor, users will have to add visual elements to their robot.